### ISSUE TITLE
Updating Member Roles CP Access Settings while using Multiple Site Manager overwrites channel access permissions

### ISSUE THREAD

**Issue by melbazany** [ExpressionEngine/ExpressionEngine#1087]
EE 6.0.4:

Using MSM (multiple site manager) (for reference, Site A and Site B). 

1. While on Site A, logged in as Super Admin: Navigating to Roles > Select "Content Administrator" > CP Access tab.
2. Scroll to "Channel Access" and Check All. **See attached screenshot here - the parent checkbox for the channel is not checked when I go to update the Channel access permissions for the Content Administrator role, but all of its children ARE checked. ** 
3. Update 'Allowed upload directory access' by checking all File upload directories for permission to access. These are all unchecked when I navigate to the Role settings.
4. Save Role Permissions.
6. Log in as a Content Administrator and access CP on Site A.
7. Attempt to view and edit an entry from all available channels on Site A and am able to do so successfully.
8. Using the CP toggle, navigate to Site B.
9. Click on Entries on the side navigation bar in the CP.
10. Click on any entry listed. Met with "Error: You are not authorized to access this page". 
11. While still on Site B, log out of Content Administrator user, and log back into the site as a Super Admin. 
12. Navigate to Roles > Select "Content Administrator" > CP Access tab.
13. Scroll to "Channel Access" and Check All. **See attached screenshot here - the parent checkbox for the channel is not checked when I go to update the Channel access permissions for the Content Administrator role, but all of its children ARE checked. ** 
14. Update 'Allowed upload directory access' by checking all File upload directories for permission to access. These are all unchecked when I navigate to the Role settings.
15. Save Role Permissions.
16. Log out of site, and log back into site as a Content Administrator.
17. Attempt to view and edit an entry from all available channels on Site B, and am able to do so successfully.
18. Using the CP toggle, navigate back to Site A.
19. Click on Entries on the side navigation bar in the CP.
20. Click on any entry listed. Met with "Error: You are not authorized to access this page".
21. Navigate back to Site B.
22. Attempt to view and edit an entry from all available channels on Site B and am able to do so successfully. Toggle back to Site B - all channel entries met with "Error: You are not authorized to access this page".

TLDR: Whichever Site from the MSM that you save the Role Permissions on _last_ is the only site that Content Administrators can access Channel Entries on. It seems as though File Directory permissions ('Allowed upload directory access') aren't being saved properly either (as they're always unchecked when I re-enter the Role Permissions settings), but I can still successfully upload to those as needed.

*See attached screenshot for referenced issue of the parent checkbox not being checked when I go to update the Channel access permissions for the Content Administrator role.*

![Screen Shot 2021-05-19 at 2 28 32 PM](https://user-images.githubusercontent.com/75503142/118864989-7fa0d880-b8ae-11eb-97a0-79d874bdf249.png)

**Comment by ops-andy** [comment]
I couldn't follow it step by step, but I was able to verify the TLDR. `Whichever Site from the MSM that you save the Role Permissions on last is the only site that Content Administrators can access Channel Entries on. `
we'll work on a fix for this ASAP.
